js/plugins/googleai/src/gemini.ts

```diff
@@ -952,11 +976,15 @@ export function defineGoogleAIModel({
         stopSequences: requestConfig.stopSequences,
         responseMimeType: jsonMode ? 'application/json' : undefined,
       };
+
+      // HACK: cast to any since those options aren't officially supported in the old SDK yet
       if (requestConfig.responseModalities) {
-        // HACK: cast to any since this isn't officially supported in the old SDK yet
         (generationConfig as any).responseModalities =
           requestConfig.responseModalities;
       }
+      if (model.info?.supports?.thinkingMode && requestConfig.thinkingConfig) {
```

Comment:
I don't think this supports check is necessary. API will ignore the fields if the model doesn't support it. And if it doesn't it better to let the user know (with and error) that the option they are setting is not supported.

```suggestion
      if (requestConfig.thinkingConfig) {
```